akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ConsumerFilterStore.scala

```diff
@@ -119,21 +151,137 @@ import akka.projection.grpc.consumer.ConsumerFilter.FilterCriteria
       case GetFilter(replyTo) =>
         replyTo ! ConsumerFilter.CurrentFilter(streamId, getState())
         Behaviors.same
+
+      case other =>
+        // only to silence compiler due to sealed trait, DdataConsumerFilterStore.InternalCommand
+        throw new IllegalStateException(s"Unexpected message [$other]")
     }
   }
 }
 
 /**
  * INTERNAL API
  */
-// FIXME
-//@InternalApi private[akka] class DdataConsumerFilterStore(
-//    context: ActorContext[ConsumerFilterStore.Command],
-//    streamId: String,
-//    notifyUpdatesTo: ActorRef[ConsumerFilterRegistry.FilterUpdated]) {
-//  import ConsumerFilterStore._
-//
-//  // FIXME
-//
-//  def behavior(): Behavior[Command] = ???
-//}
+@InternalApi private[akka] object DdataConsumerFilterStore {
+  object State {
+    val empty: State = State(Vector.empty)
+  }
+  final case class State(filterCriteria: immutable.Seq[ConsumerFilter.FilterCriteria]) extends ReplicatedData {
+    // FIXME serialization
+    type T = State
+
+    override def merge(that: State): State =
+      State(ConsumerFilter.mergeFilter(filterCriteria, that.filterCriteria))
```

Comment:
yeah, I had this in mind when implementing the mergeFilter

Reply:
It's not this simple, the removals will be lost when merging several times. Probably need to use ORSet and such.